aws/retry/middleware.go

```diff
@@ -173,3 +173,29 @@ func getRetryMetadata(ctx context.Context) (metadata retryMetadata, ok bool) {
 func setRetryMetadata(ctx context.Context, metadata retryMetadata) context.Context {
 	return context.WithValue(ctx, retryMetadataKey{}, metadata)
 }
+
+// AwsRetryMiddlewareConfig interface for retry middleware config
+type AwsRetryMiddlewareConfig interface {
+	GetRetryer() Retryer
+}
+
+// AwsRetryMiddlewares represents the Aws middleware's for Retry
+type AwsRetryMiddlewares struct {
+	attemptMiddleware       *AttemptMiddleware
+	metricsHeaderMiddleware *MetricsHeaderMiddleware
+}
+
+// AddRetryMiddlewares adds retry middleware to operation middleware stack
+func AddRetryMiddlewares(stack *smithymiddle.Stack, cfg AwsRetryMiddlewareConfig, optsFn ...func(*AwsRetryMiddlewares)) {
```

Comment:
nit shouldn't these members be exported?  Its not possible to set the values from an API client if they are not exported.

Reply:
Yes should have been exported